### PR TITLE
improve performance of noise reduction and seafloor detection

### PR DESCRIPTION
**Pull request by LOCEANlloydizard**
changes to improve performance when processing larger datasets (e.g. with Echodataflow):

- speed up Blackwell angle smoothing using uniform_filter
- reduce intermediate memory usage in background-noise estimation by combining propagation-loss terms

### CODE REVIEW

**Review by leewujung — DISMISSED** [review]
Thanks @LOCEANlloydizard , these look great!

File: echopype/clean/api.py [hunk 1]
```diff
@@ -393,11 +393,15 @@ def estimate_background_noise(
         background_noise_max = extract_dB(background_noise_max)
 
     # Compute transmission loss
-    spreading_loss = 20 * np.log10(ds_Sv["echo_range"].where(ds_Sv["echo_range"] >= 1, other=1))
-    absorption_loss = 2 * ds_Sv["sound_absorption"] * ds_Sv["echo_range"]
+    echo_range = ds_Sv["echo_range"]
+
+    transmission_loss = (
+        20 * np.log10(echo_range.where(echo_range >= 1, other=1))
+        + 2 * ds_Sv["sound_absorption"] * echo_range
```
Comment (leewujung) [inline echopype/clean/api.py]: tiny addition to note the terms
```suggestion
        20 * np.log10(echo_range.where(echo_range >= 1, other=1))  # spreading
        + 2 * ds_Sv["sound_absorption"] * echo_range  # absorption
```

**Review by leewujung — APPROVED** [review]
